CMakeLists.txt

```diff
@@ -21,13 +21,21 @@ os_set_flags()
 # macro definition located at "CMake/global_config.cmake"
 global_set_flags()
 
-if(BUILD_SHARED_LIBS)
-  message(STATUS "add_library(${LRS_TARGET} SHARED)")
-  add_library(${LRS_TARGET} SHARED "")
-endif()
+message(STATUS "add_library: ${LRS_TARGET} STATIC: ${BUILD_STATIC_LIBS} SHARED: ${BUILD_SHARED_LIBS}")
+add_library(${LRS_TARGET} OBJECT "")
 if(BUILD_STATIC_LIBS)
-  message(STATUS "add_library(${LRS_TARGET}_static STATIC)")
-  add_library(${LRS_TARGET}_static STATIC "")
+add_library(${LRS_TARGET}_static STATIC $<TARGET_OBJECTS:${LRS_TARGET}>)
+set_target_properties(${LRS_TARGET}_static PROPERTIES
+  VERSION ${REALUVC_VERSION_STRING}
+  SOVERSION ${REALUVC_VERSION_MAJOR}
+)
+endif()
+if(BUILD_SHARED_LIBS)
+add_library(${LRS_TARGET}_shared SHARED $<TARGET_OBJECTS:${LRS_TARGET}>)
+set_target_properties(${LRS_TARGET}_shared PROPERTIES
+  VERSION ${REALUVC_VERSION_STRING}
+  SOVERSION ${REALUVC_VERSION_MAJOR}
+)
 endif()
```

Comment:
Could we keep indentation on these as before, to easily see what's inside the `if` blocks?

Reply:
sure, I will add a commit